src/controller/AutoCommissioner.cpp

```diff
@@ -502,9 +502,41 @@ CHIP_ERROR AutoCommissioner::CommissioningStepFinished(CHIP_ERROR err, Commissio
             SetDAC(report.Get<RequestedCertificate>().certificate);
             break;
         case CommissioningStage::kSendAttestationRequest:
-            // These don't need to be deep copied to local memory because they are used in this one step then never again.
-            mParams.SetAttestationElements(report.Get<AttestationResponse>().attestationElements)
-                .SetAttestationSignature(report.Get<AttestationResponse>().signature);
+            if (report.Get<AttestationResponse>().attestationElements.size() > kAttestationElementsLength)
+            {
+                ChipLogError(Controller, "AutoCommissioner attestationElements buffer size %d larger than cache size %d",
+                             static_cast<int>(report.Get<AttestationResponse>().attestationElements.size()),
+                             static_cast<int>(kAttestationElementsLength));
+                mParams.SetAttestationElements(report.Get<AttestationResponse>().attestationElements);
+            }
+            else
+            {
+                memcpy(mAttestationElements, report.Get<AttestationResponse>().attestationElements.data(),
+                       report.Get<AttestationResponse>().attestationElements.size());
+                mParams.SetAttestationElements(
+                    ByteSpan(mAttestationElements, report.Get<AttestationResponse>().attestationElements.size()));
+            }
+
+            if (report.Get<AttestationResponse>().signature.size() > kAttestationSignatureLength)
+            {
+                ChipLogError(Controller,
+                             "AutoCommissioner attestationSignature buffer size %d larger than "
+                             "cache size %d",
+                             static_cast<int>(report.Get<AttestationResponse>().signature.size()),
+                             static_cast<int>(kAttestationSignatureLength));
```

Comment:
Again, please cast to unsigned and use `%u`.

Reply:
addressed